src/front/zsharp/zvisit/zgenericinf.rs

```diff
@@ -352,6 +359,7 @@ impl<'ast, 'gen, const IS_CNST: bool> ZGenericInf<'ast, 'gen, IS_CNST> {
         // 2. walk through struct def to generate constraints on inner explicit generics
         let old_sfx = std::mem::replace(&mut self.sfx, new_sfx);
         let old_gens = std::mem::replace(&mut self.gens, &strdef.generics[..]);
+        self.zgen.file_stack_push(strpath);
         for ast::StructField { ty, id, .. } in strdef.fields.iter() {
             if let Some(t) = aty_map.remove(&id.value) {
                 self.fdef_gen_ty(t, ty)?;
```

Comment:
Oh interesting. I guess we don't fully resolve types at declaration time?

Reply:
Right. Since we have to do generic inference at "call time" anyhow, I just leave all the resolution until then. I guess we could resolve "real" constants at declaration and delay resolving generics until inference time, but as far as I can tell that just adds complexity.